src/uct/ib/mlx5/dc/dc_mlx5_ep.h

```diff
@@ -690,13 +691,20 @@ uct_dc_mlx5_iface_dci_detach(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
 
     ucs_assert(!uct_dc_mlx5_iface_is_policy_shared(iface));
     ucs_assert(dci_index != UCT_DC_MLX5_EP_NO_DCI);
-    ucs_assert(iface->tx.dci_pool[uct_dc_mlx5_ep_pool_index(ep)].stack_top > 0);
 
-    if (uct_dc_mlx5_iface_dci_has_outstanding(iface, dci_index) ||
-        uct_dc_mlx5_is_hw_dci(iface, dci_index)) {
+
+    if (uct_dc_mlx5_is_hw_dci(iface, dci_index)) {
+        ep->dci    = UCT_DC_MLX5_EP_NO_DCI;
+        ep->flags &= ~UCT_DC_MLX5_EP_FLAG_TX_WAIT;
+        return 1;
```

Comment:
can we unite with lines 708-709 using goto?
we can move uct_dc_mlx5_iface_dci_schedule_release fron line 711 before lines 708-709 to achieve that